implementations/rust/ockam/ockam_vault/src/signer_impl.rs

```diff
@@ -47,6 +55,26 @@ impl Signer for Vault {
                 let sig = kp.sign(data.as_ref());
                 Ok(Signature::new(sig.to_bytes().to_vec()))
             }
+            SecretType::NistP256 => {
+                #[cfg(feature = "aws")]
+                if let Some(kms) = &self.aws_kms {
+                    if let Secret::Aws(kid) = entry.secret() {
+                        return kms.sign(kid, data).await;
+                    }
+                }
+                let key = entry.secret().try_as_key()?.as_ref();
+                cfg_if! {
+                    if #[cfg(feature = "rustcrypto")] {
+                        use p256::ecdsa::{self, signature::Signer as _};
+                        use p256::pkcs8::DecodePrivateKey;
+                        let sec = ecdsa::SigningKey::from_pkcs8_der(key).map_err(from_pkcs8)?;
+                        let sig = sec.sign(data);
+                        Ok(Signature::new(sig.to_der().as_bytes().to_vec()))
+                    } else {
+                        compile_error!("NIST P-256 requires feature `rustcrypto`")
+                    }
+                }
```

Comment:
A general question about this pattern. Couldn't this be written as follows?

```rust
cfg_if! {
    if #[cfg(feature = "aws")] { ... }
    else if #[cfg(feature = "rustcrypto")] { ... }
    else { ... }
}
```

Reply:
No, because this would not make it possible to fall back to the `rustcrypto` block from the `aws` block in case the `Secret` is not a `Secret::Aws` or because the vault is not configured for AWS. Generally, the `cfg_if!` tests are static (happen at compile time) and here we have to make some dynamic (at runtime) decisions too.

Reply:
Ok I understand, I knew I was missing something but couldn't see what. Thanks!